Parsers/ASimWebSession/Parsers/ASimWebSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,401 @@
+                                              DeviceEventCategory == 8388608, "Expanded - Unified Policy Engine (8388608)",
+                                              "Legacy Category Not Mapped"
+                                              )
+  | extend sosIPSPriority = case(ipspri == 1, "High (1)",
+                                              ipspri == 2, "Medium (2)",
+                                              ipspri == 3, "Low (3)",
+                                              ""
+                                              )
+  | extend sosAntiSpywarePriority = case(spypri == 1, "High (1)",
+                                        spypri == 2, "Medium (2)",
+                                        spypri == 3, "Low (3)",
+                                        ""
+                                        )
+  | extend
+      EventVendor = "SonicWall"
+      , EventProduct = "Firewall"
+      , DvcOs = "SonicOS"
+      , DvcOsVersion = EventProductVersion
+      , DvcIdType = "Other"
+      , Dvc = sosSerialNumber
+      , DvcDescription = DeviceProduct
+      , Rule = RuleName
+      , NetworkBytes = tolong(coalesce(toint(ReceivedBytes), 0) + coalesce(toint(SentBytes), 0))
+      , sosIPSFullString = ipscat
+      , ipscat = extract(@'^"?([a-zA-Z-\/]+)', 1, ipscat) // IPS Category/Signature
+      , sosIPSSignatureName = extract(@'[ ](.*)\S', 1, ipscat) // IPS Signature name
+      , FileSize = tolong(coalesce(FileSize, long(null)))
+      , sosAppControlFileName = extract(@'.*Filename: (.*)\"', 1, sosEventMessageDetail) // App Control Filename Logging
+      , HttpReferrer = extract(@'Referer: (.*)\"$', 1, coalesce(sosLogMsgNote, ""))
+      , sosHttpRequestMethod_ = extract(@'Command: (.\w+)', 1, coalesce(sosLogMsgNote, ""))
+      , sosCaptureATPVerdict = extract(@'Gateway Anti-Virus Status: (.*)\. ', 1, sosEventMessageDetail)
+      , sosGAVSignatureName = extract(@'Gateway Anti-Virus Alert: (.*) blocked\.', 1, sosEventMessageDetail)
+      , sosASWSignatureName = extract(@'Anti-Spyware Detection Alert: (.*)\. ', 1, sosEventMessageDetail)
+      , sosCountry = extract(@'Country Name:(.*)\"$', 1, sosEventMessageDetail)
+      , sosCFSCategoryID = extract(@'(\d+)\s', 1, coalesce(sosCFSFullString, "")) // Application Name from App Control
+      , sosCFSCategoryName = extract(@'.*-("(.*))', 1, coalesce(sosCFSFullString, "")) // Application Name from App Control
+      , sosCFSPolicyName = extract(@'Policy: (.*), Info:', 1, coalesce(sosLogMsgNote, ""))
+      , SrcZone = sosSourceZone
+      , DstZone = sosDestinationZone
+      , EventOriginalSeverity = LogSeverity
+      , Dst = DstIpAddr
+      , Src = SrcIpAddr
+      , IpAddr = SrcIpAddr
+      , EventStartTime = TimeGenerated
+      , EventEndTime = TimeGenerated
+      , EventType = "HTTPsession"
+      , EventSchemaVersion = "0.2.5"
+      , EventSchema = "WebSession"
+      , EventCount = toint(1)
+      , EventUid = _ItemId
+      , EventResultDetails = "NA"

```

Comment:
HTTP status code not coming from source?

Reply:
Unfortunately not. I reported this to the responsible team, but it has not been addressed yet. For now I'm setting it to NA.

Reply:
I will add the filter and let it remain null as you suggested in another comment.